src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -627,18 +632,22 @@ public function getEditLink($id)
      */
     private function getGridFieldItemAdjacencies(): array
     {
-        $list = $this->getGridField()->getManipulatedList();
-        $paginator = $this->getGridFieldPaginatorState();
-        if (!$paginator) {
-            return [];
+        if (!isset($this->cachedGridFieldItemAdjacencies)) {
+            $paginator = $this->getGridFieldPaginatorState();
+            if (!$paginator) {
+                $this->cachedGridFieldItemAdjacencies = [];
```

Comment:
```suggestion
                return [];
```
Caching this empty result doesn't actually improve anything, so we may as well directly return it.

Reply:
Updated